lib/broadway_kafka/acknowledger.ex

```diff
@@ -93,7 +93,7 @@ defmodule BroadwayKafka.Acknowledger do
   end
 
   # Bump latest offset
-  defp update_offsets([offset | offsets], current, all_pending) when current + 1 == offset do
+  defp update_offsets([offset | offsets], current, all_pending) when current + 1 <= offset do
```

Comment:
I believe we cannot do this because it means we will ack a missing offset (say 12 in the examples below) when the last offset arrives. When you have compactation, do you have gaps on the offsets? For example, can you have 12, 13, 15, 16... or does it only have an initial gap and then everything else is contiguous?

Reply:
That is what I was worried about.  When I was logging out the offsets during my debugging it did appear that when compaction happened there was a gap at the beginning I was seeing:

```
[10, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125,
 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141,
 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157,
 158, 159, ...]
```

so from the offset of 10, it jumped to 111 and then appears to be contiguous

Reply:
I think this will happen in case of failures. If a message fails in handle_message, Broadway will ack the failed messages before it acks the successful ones in handle_batch. So if the messages can have gaps in them, then we will need to change this approach to work with sets (i.e. instead of keeping an offset, we keep a set of all offsets we know and we only ack when the set is empty).